### PR TITLE
Un-removes Juggernaut/Marauder dash, and "properly" fixes server-crashing bug, and also several related minor bugs

### PR DESCRIPTION
**Pull request by Exxion**
* Reverts #20672
* Fixes all grilles dropping *rods* when hit by thrown mechs/big juggernauts as opposed to whatever they're supposed to drop
* Fixes grilles flipping back and forth between damaged and undamaged states every time `healthcheck()` was called if they were at exactly 1/4 of their starting health (in which case they now stay damaged)
* Fixes cult grilles being impassable even when broken
  * This was what was causing the server crash, so also fixes that

### CODE REVIEW

File: code/game/objects/structures/grille.dm [hunk 1]
```diff
@@ -295,7 +295,7 @@
 /obj/structure/grille/cult/Cross(atom/movable/mover, turf/target, height = 1.5, air_group = 0)
 	if(air_group || !broken)
 		return 0 //Make sure air doesn't drain
-	..()
+	return ..()
```
Comment (Exxion) [inline code/game/objects/structures/grille.dm:298]: This is the line that fixes the crash

**Comment by SonixApache** [comment]
Ramming a grille with a mech used to drop *rods*? Like, *immovable rods*?

**Comment by Exxion** [comment]
Metal rods
Not all kinds of grille are supposed to drop metal rods

**Comment by Exxion** [comment]
Oh by the way, this allows for clever airflow stuff with cult grilles
Broken ones still block air, but not objects
Get a cultify artifact and build an open-"air" space bar by surrounding it in broken cult grilles

**Pull request merged**